anndata/_core/anndata.py

```diff
@@ -1295,11 +1297,13 @@ def transpose(self) -> "AnnData":
                 "which is currently not implemented. Call `.copy()` before transposing."
             )
 
-        def t_csr(m: sparse.spmatrix) -> sparse.csr_matrix:
-            return m.T.tocsr() if sparse.isspmatrix_csr(m) else m.T
+        # TODO: Was this imoprtant?
+        # def t_csr(m: sparse.spmatrix) -> sparse.csr_matrix:
+        #     return m.T.tocsr() if sparse.isspmatrix_csr(m) else m.T
 
         return AnnData(
-            X=t_csr(X) if X is not None else None,
+            X=_safe_transpose(X) if X is not None else None,
+            # X=t_csr(X) if X is not None else None,
```

Comment:
@flying-sheep do you remember why we would convert to csr here? It seems like an odd choice considering transpose is "free" if we don't insist on the result type.

Reply:
I think older versions of scanpy or anndata assumed csr, but I don’t really know

Reply:
It seems fine... But part of me feels like we should do a deprecation.

Tbh I think transposing is pretty rare. Maybe we change it with a release candidate and if anyone complains we do a deprecation?